src/nvim/profile.c

```diff
@@ -176,22 +169,31 @@ bool profile_equal(proftime_T tm1, proftime_T tm2) FUNC_ATTR_CONST
   return tm1 == tm2;
 }
 
-/// Calculates the sign of a 64-bit integer.
+/// Converts a proftime_T value `tm` to a signed integer.
 ///
-/// @return -1, 0, or +1
-static inline int sgn64(int64_t x) FUNC_ATTR_CONST
+/// @return signed representation of the given time value
```

Comment:
Sorry for the extremely delayed review. I think the CL is good, but I had one more thought.

Being exposed to Go a lot made me think that I would solve it with two distinct types:

  - A type for points in time (proftime_T).
  - A type for the subtraction of two points in time (duration_T or something else). This could be unsigned for negative durations. `profile_sub` would take two `proftime_T`'s and return a `duration_T`.

(It's been a while, but I think C's automatic type conversion might make accidentally assigning one to the other far too easy, and it's not worth defining two types.)

At first sight, they're semantically quite different. One is an absolute point in time (though we don't know the starting point), and the other is a duration.

On second look, there's also something to be said for keeping them as the same type. Essentially, proftime_T (as defined above) is the duration since the processor was started. When subtracting two proftime_T's, it's just rebasing the starting point to the second operand.

I guess what I'm trying to say is: the way it is right now is IMHO a decent tradeoff, as C won't provide too much protection either way (Go would complain if one doesn't cast, even for the same underlying type). 

The only thing I'd change is to document that `profile_signed` is meant to be used `proftime_T`s that are (perhaps indirectly) the result of `profile_sub` calls.

The comment itself is not that important, maybe if you ever pass over the code again.

Reply:
Thanks. I will update the comment.

> Essentially, proftime_T (as defined above) is the duration since the processor was started.

That was my conclusion as well, though I didn't inspect every call site in the codebase.